app/graphql/types/query_type.rb

```diff
@@ -4,58 +4,34 @@ module Types
   class QueryType < Types::BaseObject
     description 'The query root of this schema'
 
-    field :user_id_query, [Types::EventType], null: false do
-      description 'returns all events with user_id'
-      argument :userId, ID, required: true
+    field :stats_count, [Types::EventType], null: false do
+      description 'returns counts of events in category grouped into period with length of interval (i.e. "1 day")'
+      argument :interval, String, required: true
+      argument :event_type, String, required: true
+      argument :user_id, ID, required: false
+      argument :project_id, ID, required: false
+      argument :workflow_id, ID, required: false
     end
 
-    def user_id_query(kwargs, searcher=Searchers::Complete)
-      if user_permission?(kwargs[:user_id])
-        searcher.new.search(**kwargs)
-      else
+    def stats_count(kwargs, searcher=Searchers::Bucket)
+      if site_wide_search?(kwargs) && (not context[:admin])
```

Comment:
This can probably be improve so there is only one raise error. Should I separate it out into a method?

Reply:
extract to method here can help with clarity of the behaviour, especially if its a well named method. Instead of interpretting the boolean logic i can read the method name to understand what its meant to be doing... 

My assumption for the behaviour here is to only allow admins to run the site wide search?